### PR TITLE
Add TF-Vision RetinaNet test

### PR DESCRIPTION
**Pull request by A9isha**
Add TF `vision-retinanet` test

### CODE REVIEW

File: tests/tensorflow/nightly/vision-retinanet.libsonnet [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      },
+    },
+  },
+  local functional_v2_8 = common.Functional {
+    scriptConfig+: {
+      paramsOverride+: {
+        task: {
+          annotation_file: '$(COCO_DIR)/instances_val2017.json',
+          train_data: {
+	    global_batch_size: 64,
```
Comment (will-cromar) [inline tests/tensorflow/nightly/vision-retinanet.libsonnet]: Add this to `v2_8` instead:

```
      paramsOverride+: {
        task: {
          train_data: {
	    global_batch_size: 64,
          },
        },
      },
```

This way, the batch size change takes effect for both the functional and convergence tests, and you don't have to have two copies of `functional`.

File: tests/tensorflow/nightly/vision-retinanet.libsonnet [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    PreemptibleTpu:: {
+      tpuSettings+: {
+        preemptible: false,
+      },
+    },
+  },
+  local functional = common.Functional {
+    scriptConfig+: {
+      paramsOverride+: {
+        task: {
```
Comment (will-cromar) [inline tests/tensorflow/nightly/vision-retinanet.libsonnet:37]: Move these common fields to `retinanet` instead:

```
      paramsOverride+: {
        task: {
          annotation_file: '$(COCO_DIR)/instances_val2017.json',
	  train_data: {
            input_path: '$(COCO_DIR)/train*',
          },
	  validation_data: {
            input_path: '$(COCO_DIR)/val*',
          },
        },
```

The input data is common to both `functional` and `convergence`, but right now the convergence test doesn't point to any input data.

File: tests/tensorflow/nightly/vision-retinanet.libsonnet [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+local tpus = import 'templates/tpus.libsonnet';
+local utils = import 'templates/utils.libsonnet';
+
+{
+  local retinanet = common.TfVisionTest + {
+    modelName: 'vision-retinanet',
+    scriptConfig+: {
+      experiment: 'retinanet_resnetfpn_coco',
+    },
+    PreemptibleTpu:: {
```
Comment (will-cromar) [inline tests/tensorflow/nightly/vision-retinanet.libsonnet:28]: Remove this `PreemptibleTpu` block. It probably doesn't do what you want it to, and we want to control TPU preemptibility in common templates.

**Review by will-cromar — CHANGES_REQUESTED** [review]

File: tests/tensorflow/nightly/vision-retinanet.libsonnet [hunk 4]
```diff
@@ -0,0 +1,107 @@
+// Copyright 2021 Google LLC
```
Comment (will-cromar) [inline tests/tensorflow/nightly/vision-retinanet.libsonnet:1]: 2022

File: tests/tensorflow/nightly/vision-retinanet.libsonnet [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        },
+      },
+    },
+  },
+  local val_32 = {
+    scriptConfig+: {
+      paramsOverride+: {
+        task+: {
+          validation_data+: {
+            global_batch_size: 32,
```
Comment (will-cromar) [inline tests/tensorflow/nightly/vision-retinanet.libsonnet:69]: I would create a common TPU subclass where you compute the validation batch size according to the number of replicas (i.e. `config.accelerator.replicas`). See this example:

https://github.com/GoogleCloudPlatform/ml-testing-accelerators/blob/7dec2008823f97ec67a4ab444a8975d4aa2e1ad2/tests/tensorflow/r2.8/maskrcnn.libsonnet#L91-L93

**Review by will-cromar — CHANGES_REQUESTED** [review]
Otherwise LGTM

**Comment by will-cromar** [comment]
Can you also rename this file `vision-coco` to align with `vision-imagenet`? We may want to add more tests that use the same dataset

File: tests/tensorflow/nightly/vision-retinanet.libsonnet (hunk 4 above)
Comment (A9isha) [reply tests/tensorflow/nightly/vision-retinanet.libsonnet]: done

**Review by will-cromar — APPROVED** [review]
LGTM. Great work!

**Pull request merged**